controllers/velero.go

```diff
@@ -531,3 +530,31 @@ func (r *VeleroReconciler) getVeleroResourceReqs(velero *oadpv1alpha1.Velero) co
 
 	return ResourcesReqs
 }
+
+// For later: Move this code into validator.go when more need for validation arises
+// TODO: if multiple default plugins exist, ensure we validate all of them.
+// Right now its sequential validation
+func (r *VeleroReconciler) ValidateVeleroDeployment(log logr.Logger) (bool, error) {
+	velero := oadpv1alpha1.Velero{}
+	if err := r.Get(r.Context, r.NamespacedName, &velero); err != nil {
+		return false, err
+	}
+
+	secretName := "nosecret"
```

Comment:
I would initialize this as an empty string `secretName := ""`

Reply:
This was my initial PR. I had an empty string and for the condition, I equated the length of the string to zero. It failed on Openshift clusters < 4.7 but passed in my local cluster (which is running 4.8.x). Will try what you have suggested. Thanks @dymurray